client/src/components/JurisdictionAdmin/DownloadBatchTallySheetsButton/DownloadBatchTallySheetsButton.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect } from 'react'
+import { AnchorButton } from '@blueprintjs/core'
+import { BlobProvider, Document } from '@react-pdf/renderer'
+import { toast } from 'react-toastify'
+
+import BatchTallySheet from './BatchTallySheet'
+import useContestsJurisdictionAdmin from '../useContestsJurisdictionAdmin'
+import { useBatches } from '../useBatchResults'
+
+const AnchorButtonErrorState = (): JSX.Element => {
+  useEffect(() => {
+    toast.error('Error preparing batch tally sheets for download')
+  }, [])
+
+  return (
+    <AnchorButton disabled icon="th">
+      Download Batch Tally Sheets
+    </AnchorButton>
+  )
+}
+
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  jurisdictionName: string
+  roundId: string
+}
+
+const DownloadBatchTallySheetsButton = ({
+  electionId,
+  jurisdictionId,
+  jurisdictionName,
+  roundId,
+}: IProps): JSX.Element | null => {
+  const batchesResp = useBatches(electionId, jurisdictionId, roundId)
+  const contests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+
+  if (!batchesResp || !batchesResp.data || !contests || contests.length === 0) {
+    return null
+  }
+
+  const { batches } = batchesResp.data
+  // Batch comparison audits only support a single contest
+  const contest = contests[0]
+
+  return (
+    <BlobProvider
+      document={
+        <Document title="batch-tally-sheets.pdf">
+          {batches.map(b => (
+            <BatchTallySheet
+              auditBoardName={b.auditBoard ? b.auditBoard.name : ''}
+              batchName={b.name}
+              choices={contest.choices}
+              jurisdictionName={jurisdictionName}
+              key={b.id}
+            />
+          ))}
+        </Document>
+      }
+    >
+      {({ error, loading, url }) => {
+        if (error) {
+          return <AnchorButtonErrorState />
+        }
+        return (
+          <AnchorButton
+            href={url || undefined}
+            icon="th"
+            loading={loading}
+            target="_blank"
+          >
+            Download Batch Tally Sheets
+          </AnchorButton>
+        )
+      }}
```

Comment:
So this UX is a little different from the existing buttons, which generate their PDFs on click. Pre-generating seems nice because I assume it will be faster to download on click (though I wonder if it will impact page rendering performance at all - did you test on an election with many sampled batches?)

I don't know that it makes sense to make pre-rendering transparent to the user. I think it makes more sense for them to think that nothing happens until they click the button, since they might load the page and not choose to download those PDFs.

A few thoughts:
- Is it possible to not pre-generate the PDFs and only generate them on button click?
- Could we pre-generate in the background, but then if they click before we're done, have the button spin then? (Or if there was an error, toast the error then)

Reply:
Great call re `setImmediate`! Will incorporate that!

And yup, I've moved the error handling into the `onClick` in my actual cleaned up code

Also agree that caching is probably riskier than it's worth for now

Reply:
And here! https://github.com/votingworks/arlo/pull/1475/commits/ccdf21d58ca16d2c0d6d2a71522ee7f10236c1a9

With the minimal update to get tests working in https://github.com/votingworks/arlo/pull/1475/commits/b484c60659b9bdb669c4a4bdf8aa574e8b46a5ba. Still trying to make some improvements there (https://github.com/votingworks/arlo/pull/1475#discussion_r870626541)